web/src/main/kotlin/com/example/ajaxproject/service/EmailSenderServiceImpl.kt

```diff
@@ -9,30 +9,37 @@ import org.slf4j.LoggerFactory
 import org.springframework.mail.javamail.JavaMailSender
 import org.springframework.mail.javamail.MimeMessageHelper
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
 
 @Service
 internal class EmailSenderServiceImpl(private val javaMailSender: JavaMailSender) : EmailSenderService {
 
-    override fun send(emailDTO: EmailDTO): SendEmailResponse {
-        return runCatching {
+    override fun send(emailDTO: EmailDTO): Mono<SendEmailResponse> {
+        return Mono.defer {
             javaMailSender.send(generateMailMessage(emailDTO))
```

Comment:
Interesting question)
Send function itself is not blocking, but the javaMailSender.send(generateMailMessage(emailDTO)) call is likely a blocking operation because it's sending an email, which typically involves network operations that can block the calling thread
So , i refactored this code in reactive way